app/Console/Commands/AdsGetTxIn.php

```diff
@@ -0,0 +1,158 @@
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+namespace Adshares\Adserver\Console\Commands;
+
+use Adshares\Ads\AdsClient;
+use Adshares\Ads\Exception\CommandException;
+use Adshares\Ads\Util\AdsConverter;
+use Adshares\Adserver\Exceptions\ConsoleCommandException;
+use Adshares\Adserver\Models\AdsTxIn;
+use Adshares\Adserver\Models\Config;
+use Illuminate\Console\Command;
+use Illuminate\Database\QueryException;
+
+class AdsGetTxIn extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ads:get-tx-in';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Searches blockchain user log for incoming transfers';
+
+    /**
+     * Command ended without error
+     */
+    const EXIT_CODE_SUCCESS = 0;
+
+    /**
+     * Command ended with an error
+     */
+    const EXIT_CODE_ERROR = 1;
+
+    /**
+     * Create a new command instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     * @param AdsClient $adsClient
+     * @return int
+     */
+    public function handle(AdsClient $adsClient)
+    {
+        /** @var $from Config */
+        $from = Config::where('key', Config::ADS_LOG_START)->first();
+        if (null === $from) {
+            $from = Config::create(['key' => Config::ADS_LOG_START, 'value' => '0']);
+        }
+        $timestamp = intval($from->value);
+        $dateFrom = (new \DateTime)->setTimestamp($timestamp);
+        try {
+            $response = $adsClient->getLog($dateFrom);
+        } catch (CommandException $exc) {
+            $this->error('Cannot get log');
+            return self::EXIT_CODE_ERROR;
+        }
+        $log = $response->getLog();
+
+        if (count($log) > 0) {
+            $txsCount = $this->parse($log);
+
+            try {
+                $from->value = $this->getLastEventTime($log);
+                $from->save();
+            } catch (ConsoleCommandException $exc) {
+                $this->error('Cannot get time of last event');
+            }
+        } else {
+            $txsCount = 0;
+        }
+        $this->info("Number of added txs: ${txsCount}");
+        return self::EXIT_CODE_SUCCESS;
+    }
+
+    /**
+     * Returns time of last event in log.
+     * @param array $log log
+     * @return int time of last event
+     * @throws ConsoleCommandException if log array is empty
+     */
+    private function getLastEventTime(array $log): int
+    {
+        $count = count($log);
+        $lastEventIndex = $count - 1;
+        if ($count > 0) {
+            $time = $log[$lastEventIndex]['time'];
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$time". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)